src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -89,9 +90,15 @@ public class GenerateMapController extends NodeController<Pane> {
       IntStream.range(2, 17).boxed().collect(Collectors.toList()));
   private final FilteredList<Integer> selectableSpawnCounts = new FilteredList<>(validSpawnCount);
   public Spinner<Integer> numTeamsSpinner;
+  private BooleanBinding customizationAllowed;
 
   @Override
   protected void onInitialize() {
+    customizationAllowed = previousMapName.textProperty().isNotEmpty()
+                                          .or(generationTypeComboBox.valueProperty()
+                                                                    .isNotEqualTo(GenerationType.CASUAL))
+                                          .or(commandLineArgsText.textProperty().isNotEmpty());
+
```

Comment:
This is better as a private final BooleanProperty that then gets bound in onInitialize, also as is wouldn't the variable be better named noCustomizatonAllowed?

Reply:
Ehm, not sure if I get you on the first part. if I make property final, it won't go as it will fall into category to be autowired (there is no bean to be autowire) additionally, it depends on previousMapName textField. So therefore no way to make it final. Alternative is to make it local and then pass it as param. Or am I thinking about something else? Is there Object BooleanBinding that can be specified at later time?

hmm, yeah or forbidCustomization / disableCustomization? In general, I'm lost in these logical operators and then their binding to field/property....

Reply:
The property wouldn't be auto wired because we should initialize it at declaration so just 'private final BooleanProperty disableCustomization = new Simple BooleanProperty();'

And then in onInit we would just disableCustomization.bind(...).

This way the property itself is final and we don't have to worry about it being reassigned. And it fits the pattern in most of the other controllers

Reply:
Ouch, forgot about SimpleBooleanProperty ... jobs done.

Anyway, are these all <type>Property() classes used nowadays in projects? I've never came upon them till now, though I'm working only on web-applications ...

Reply:
They are used pretty heavily in JavaFX and are specific to it. But JavaFX is a desktop application library so if you have mainly done web application development you probably don't see them very often.

But mainly they are JavaFX's general solution to reactivity and event listeners. Somewhat like promises but with change detection and the like that you expect from fields.